src/aiida/engine/processes/calcjobs/tasks.py

```diff
@@ -426,13 +428,19 @@ async def do_kill():
             transport = await cancellable.with_interrupt(request)
             return execmanager.kill_calculation(node, transport)
 
+    if force_kill:
+        logger.warning(f'Process<{node.pk}> has been force killed! this may result in orphaned jobs.')
+        raise plumpy.process_states.ForceKillInterruption('Force killing CalcJob')
     try:
         logger.info(f'scheduled request to kill CalcJob<{node.pk}>')
         result = await exponential_backoff_retry(do_kill, initial_interval, max_attempts, logger=node.logger)
+    # Note: any exception raised here, will result in the process being excepted. not killed!
+    # There for it can result in orphaned jobs!
     except plumpy.process_states.Interruption:
+        logger.warning(f'killing CalcJob<{node.pk}> excepted, the job might be orphaned.')
         raise
     except Exception as exception:
-        logger.warning(f'killing CalcJob<{node.pk}> failed')
+        logger.warning(f'killing CalcJob<{node.pk}> excepted, the job might be orphaned.')
```

Comment:
Don't think this warning is correct. When a `TransportTaskException` is raised, the `Waiting.execute` method catches it and pauses the process.

Reply:
If I'm not wrong, what you are referring to is already in my test scenario 
(see here `test_process.py::test_process_kill` # 10)
And is being passed, meaning the job is being EXCEPTED, not paused..
So I'm not sure..

Reply:
I don't understand. The exception being caught here is thrown by `do_kill` so it is the kill operation (i.e. getting the transport or using it to call the kill command of the scheduler) that excepted, not the process itself. The exception is then reraised as a `TransportTaskException` which is caught in `Waiting.execute` and rethrown as `PauseInterruption` on line 571, which means the process is going into pause. So how does this mean the process is already excepted?

Reply:
I see.
Just added one more test, and I'm now explicitly patching and returning `TransportTaskException` from `task_kill_job`. 
We still receive `EXCEPTED` regardless of whether the process was running normally or was stuck in EBM. 
Haven't figured out why yet.